### PR TITLE
T5 compile compatibilty

### PR DESCRIPTION
**Pull request by zucchini-nlp**
# What does this PR do?

Same as https://github.com/huggingface/transformers/pull/33754, I accidentally force pushed to wrong branch and the PR got closed 🙃

### CODE REVIEW

File: tests/models/t5/test_modeling_t5.py
Comment (gante) [inline tests/models/t5/test_modeling_t5.py:1645]: Is this a new failure? 

If yes: can you confirm a) the generated tokens are different b) compare the logprobs of the different tokens before/after the change, to see whether it can be explained by tiny fluctuations?

If no: can you add a comment with the PR that causes the difference? 🙏

**Review by gante — APPROVED** [review]
Thank you for this refactor 🙏 

I've left a question for us to double-check, but approving since it looks good to me.

______
⚠️ before merging, make sure to run the following slow tests to confirm that there are no regressions vs `main`: {all touched models} + {llama, whisper, caches}

**Review by ArthurZucker — APPROVED** [review]
NICE! 🔥 thanks for adding the compile integration tests 🤗

File: tests/models/t5/test_modeling_t5.py
Comment (zucchini-nlp) [reply tests/models/t5/test_modeling_t5.py:1647]: Added a comment, the regression was caused by changes in the tokenization. So nothing breaks in terms of modeling or generation. Not sure if we have to the fix in tokenization was intended to break (i.e. fix this) or we need another PR to fix tokenization back, so I'll leave a comment for future us

**Comment by zucchini-nlp** [comment]
Will merge this later today if we are all okay. The comment for tokenization is added within code with a TODO for us, in case it is not a bug we can remove the comment later

**Pull request merged**
